processor/snapshotprocessor/README.md

```diff
@@ -0,0 +1,57 @@
+# Snapshot Processor
+
+The snapshot processor is used in custom distributions of the collector to provide snapshot functionality in BindPlane. It is not currently included in the official `bindplane-agent`.
+## Supported pipelines
+
+- Logs
+- Metrics
+- Traces
+
+## How it works
+
+1. The user configured the processor in the pipeline.
+2. Whenever telemetry passes through the pipeline, it is copied and stored temporarily in an internal buffer.
+3. An OpAMP server is able to use a custom message to request the contents of the internal buffer, in order to view a snapshot of the telemetry flowing through the collector.
+
+## Configuration
+
+| Field   | Type   | Default | Required | Description                                                            |
+|---------|--------|---------|----------|------------------------------------------------------------------------|
+| enabled | bool   | `true`  | `false`  | Whether the snapshot processor is enabled or not.                      |
+| opamp   | string | `opamp` | `false`  | Specifies the name of the opamp extension for sending custom messages. |
```

Comment:
What is the purpose if there is no opamp extension?

Reply:
There is none. I think I was treating the required field as "you need to specify this field" and not "you can't leave this field explicitly empty". I think it makes more sense to mark as required.